### PR TITLE
Extend healthcheck endpoint with detailed query param

### LINKED ISSUE #16: Extended health check endpoint
Summary:

Extend `/v1/health` to include LLM provider status, GEE auth, model availability, disk usage.

Acceptance Criteria:

- [ ]  `?detailed=true` query param returns extended info
- [ ]  Without param, returns current simple response (backwards compatible)
- [ ]  Response fast (<500ms)

### PR DESCRIPTION
**Pull request by code1715**
## Description

Extended the `/v1/health` endpoint to provide comprehensive system diagnostics via a new `?detailed=true` query parameter.

## Related Issue

Fixes [#16](https://github.com/IPT-MMDA/DT4LC-project/issues/16)

## Type of Change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update
- [ ] Refactoring (no functional changes)
- [ ] CI/CD changes

## Changes Made

- Added a `detailed` boolean query parameter to the `/v1/health` endpoint (defaults to `False`).
- Implemented logic to safely collect and return LLM provider availability, GEE authentication readiness, model registry status, and local disk usage (uploads, caches, exports).
- Updated `docs/API.md` to document the new `?detailed=true` parameter and its corresponding JSON response schema.

## Checklist

- [x] My code follows the project's style guidelines
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published

## Additional Notes

All detailed checks are designed to be extremely fast (reading configuration states, environment variables, or simple directory byte counts) without making any live HTTP requests to external APIs, ensuring the endpoint remains performant.

### CODE REVIEW

File: server/app.py [hunk 1]
```diff
@@ -48,10 +65,1576 @@ async def lifespan(app: FastAPI):
...
+    except Exception as exc:
+        models_info["error"] = str(exc)
+
+    # Disk usage (cache / uploads / exports)
+    disk_usage: dict[str, Any] = {}
+    try:
+        def _dir_size_bytes(p: Path) -> int:
+            if not p.exists():
+                return 0
+            return sum(f.stat().st_size for f in p.rglob("*") if f.is_file())
```
Comment (roaffix) [inline server/app.py]: Doesn't seem like "extremely fast" according to your description :)
This recursive I/O may be costly in terms of local resources

File: server/app.py [hunk 2]
```diff
@@ -1,44 +1,61 @@
...
-from .routes.gee import router as gee_router
-from .routes.health import router as health_router
-from .routes.jobs import router as jobs_router
-from .routes.tiles import router as tiles_router
-
+from dta.config import UPLOADS_PATH
+from dta.dti.coe.orchestrator import orchestrate
+from dta.dti.coe.llm.config import get_default_config
+from dta.dti.coe.llm.router import LLMRouter
+from dta.dti.data_sources.gee_common import _ee_initialized
```
Comment (roaffix) [inline server/app.py]: You break encapsulation here. Revisit your approach or add something like public `is_initialized()` accessor in `gee_common` file. And, probably, consider doing a real check

File: server/app.py [hunk 3]
```diff
@@ -48,10 +65,1576 @@ async def lifespan(app: FastAPI):
...
+        llm_cfg = get_default_config()
+
+        router = LLMRouter.from_config(llm_cfg)
+        for provider in router.providers:
+            llm_providers.append({
+                "name": provider.name,
+                "model": provider.model,
+                "available": provider.is_available(),
+            })
+    except Exception as exc:
```
Comment (roaffix) [inline server/app.py]: Too generic. You sometimes miss the real bugs. As for the diagnostics endpoint added - that's not really that defensible.

**Review by roaffix — CHANGES_REQUESTED** [review]
The detailed response leaks the absolute filesystem paths, GEE project ID, whether a service account is configured, a full list of installed model IDs (not critical, but still), and the disk usage directory. This is completely fine for local development, but if `/v1/health` is reachable from the internet (and we here explicitly expose it) - its a free reconnaissance endpoint.
So, we really need to require auth, or at least restrict `?detailed=true` to requests from internal networks.

As for a minor catch - `docs/API.md` changes its look, but the Pydantic model is not. I'd keep them in sync.

Apart from stated changes, this PR is missing tests, which is also crucial for any backend changes

**Comment by code1715** [comment]
@roaffix Hello again :)

Addressed all review comments:
- Moved `?detailed=true` to `server/routes/health.py` (where the endpoint now lives after refactor) and restricted it to localhost-only via `request.client.host` check - returns 403 for any non-local request. Full auth would require introducing new infrastructure from scratch.
- Removed `project_id` and filesystem paths from the response to minimize information exposure even on localhost.
- Added `is_initialized()` public accessor in `gee_common.py` instead of importing `_ee_initialized` directly; also added a real connectivity check via `ee.Number(1).getInfo()` instead of just reading the cached flag.
- Replaced `Path.rglob()` with `os.scandir()` for disk size calculation.
- Added Pydantic response models (`HealthResponse`, `GEEStatus`, `DiskUsage`, etc.) in `server/schemas.py` to keep the code and `docs/API.md` in sync.
- Added 22 tests in `tests/test_health_detailed.py` covering the localhost gate, all diagnostic collectors, and edge cases (GEE connection failure, missing env vars, empty dirs).

**Review by roaffix — APPROVED** [review]
LGTM
